oggm/core/preprocessing/climate.py

```diff
@@ -136,6 +136,230 @@ def process_custom_climate_data(gdir):
+    year = np.array([t.year for t in temp.time.values])  # selects for each point in time the year
+    month = np.array([t.month for t in temp.time.values])  # selects for each point in time the month
+
+    year = np.array([t.year for t in precp.time.values])  # selects for each point in time the year
+    month = np.array([t.month for t in precp.time.values])  # selects for each point in time the month
+
+    time= temp.time
+
+    y0=year[0]
+    y1=year[-1]
+
+    ny, r = divmod(len(temp.time), 12)
+    assert r == 0
+
+    # gradient default params
+    use_grad = cfg.PARAMS['temp_use_local_gradient']
+    def_grad = cfg.PARAMS['temp_default_gradient']
+    g_minmax = cfg.PARAMS['temp_local_gradient_bounds']
+
+    lon = gdir.cenlon
+    lat = gdir.cenlat
+
+    # This is guaranteed to work because I prepared the file (I hope)
+    ncclim.set_subset(corners=((lon, lat), (lon, lat)), margin=1)
+
+    # get climatology data
+    loc_hgt = ncclim.get_vardata('elev')
+    loc_tmp = ncclim.get_vardata('temp')
+    loc_pre = ncclim.get_vardata('prcp')
+    loc_lon = ncclim.get_vardata('lon')
+    loc_lat = ncclim.get_vardata('lat')
+
+    # see if the center is ok
+    if not np.isfinite(loc_hgt[1, 1]):
+        # take another candidate where finite
+        isok = np.isfinite(loc_hgt)
+
+        # wait: some areas are entirely NaNs, make the subset larger
+        _margin = 1
+        while not np.any(isok):
+            _margin += 1
+            ncclim.set_subset(corners=((lon, lat), (lon, lat)), margin=_margin)
+            loc_hgt = ncclim.get_vardata('elev')
+            isok = np.isfinite(loc_hgt)
+        if _margin > 1:
+            log.debug('%s: I had to look up for far climate pixels: %s',
+                      gdir.rgi_id, _margin)
+
+        # Take the first candidate (doesn't matter which)
+        lon, lat = ncclim.grid.ll_coordinates
+        lon = lon[isok][0]
+        lat = lat[isok][0]
+        # Resubset
+        ncclim.set_subset()
+        ncclim.set_subset(corners=((lon, lat), (lon, lat)), margin=1)
+        loc_hgt = ncclim.get_vardata('elev')
+        loc_tmp = ncclim.get_vardata('temp')
+        loc_pre = ncclim.get_vardata('prcp')
+        loc_lon = ncclim.get_vardata('lon')
+        loc_lat = ncclim.get_vardata('lat')
+
+    assert np.isfinite(loc_hgt[1, 1])
+    isok = np.isfinite(loc_hgt)
+    hgt_f = loc_hgt[isok].flatten()
+    assert len(hgt_f) > 0.
+    ts_grad = np.zeros(12) + def_grad
+    if use_grad and len(hgt_f) >= 5:
+        for i in range(12):
+            loc_tmp_mth = loc_tmp[i, ...][isok].flatten()
+            slope, _, _, p_val, _ = stats.linregress(hgt_f, loc_tmp_mth)
+            ts_grad[i] = slope if (p_val < 0.01) else def_grad
+    # ... but dont exaggerate too much
+    ts_grad = np.clip(ts_grad, g_minmax[0], g_minmax[1])
+    # convert to timeserie and hydroyears
+    ts_grad = ts_grad.tolist()
+    ts_grad = ts_grad[9:] + ts_grad[0:9]
+    ts_grad = np.asarray(ts_grad * ny)
+
```

Comment:
Instead of doing all these things above, you could actually assume that a climate file is already present in the directory (which it should), and use this data as reference.

This is actually better because it avoids code repetition, and makes sure that we agree upon the actual climate data. 

Also you don't need to take the three closest pixels anymore, just the actual time series